src/user-event/clear.ts

```diff
@@ -0,0 +1,60 @@
+import { ReactTestInstance } from 'react-test-renderer';
+import { ErrorWithStack } from '../helpers/errors';
+import { isHostTextInput } from '../helpers/host-component-names';
+import { isPointerEventEnabled } from '../helpers/pointer-events';
+import { EventBuilder } from './event-builder';
+import { UserEventInstance } from './setup';
+import { dispatchEvent, wait } from './utils';
+import { emitTypingEvents } from './type/type';
+
+export async function clear(
+  this: UserEventInstance,
+  element: ReactTestInstance
+): Promise<void> {
+  if (!isHostTextInput(element)) {
+    throw new ErrorWithStack(
+      `clear() works only with host "TextInput" elements. Passed element has type "${element.type}".`,
+      clear
+    );
+  }
+
+  // Skip events if the element is disabled
+  if (element.props.editable === false || !isPointerEventEnabled(element)) {
+    return;
+  }
+
+  // 1. Enter element
+  dispatchEvent(element, 'focus', EventBuilder.Common.focus());
+
+  // 2. Select all
+  const previousText = element.props.value ?? element.props.defaultValue ?? '';
+  const selectionRange = {
+    start: 0,
+    end: previousText.length,
```

Comment:
since it starts at 0, shoud not the end be `previousText.length -1` ?

Reply:
No this is actually fine, `0` is before first character, `1` is after first character, so `n` would be after `n` characters.

Reply:
oh yes understood